server/document_downloader.py

```diff
@@ -1,64 +1,127 @@
+import datetime
+import lxml.etree as ET
+import re
+import requests
+
 import cachetools.func
 from settings import CACHE_TTL_SECS, CACHE_MAX_SIZE
-import requests
 from flask_restful import abort
 from utils import server_log
 
+
 @cachetools.func.ttl_cache(ttl=CACHE_TTL_SECS, maxsize=CACHE_MAX_SIZE)
 def get_doi_content(user_doi):
     """
     This function is designed to render the paper-journal
-    network for the user given a biorxiv doi
+    network for the user given identifier.
+
+    Code Grabs the latest preprint version to generate results.
+
     Args:
-        user_doi - a biorxiv doi that grabs the most current version of a preprint
+        user_doi - user provided biorxiv/medrxiv doi or arxiv id
     """
 
-    # Try pinging biorxiv server first
-    content = ping_biorxiv_or_medrxiv(user_doi, server="biorxiv")
-    doc_url = f"http://biorxiv.org/content"
+    # create flag if xml is found
+    xml_found = False
+
+    # Check to see if user_doi is a biorxiv/medrxiv doi
+    # if not then resort to searching for arxiv
+    bio_med_rxiv = re.search(r"10\.1101\/", user_doi) is not None
```

Comment:
Is 10.1101 a thing in all doids? Or just something that will show up on bio/medrxiv posts but not arxiv?

Reply:
>s 10.1101 a thing in all doids? Or just something that will show up on bio/medrxiv posts but not arxiv?

10.1101 is a special characteristic of bio/medrxiv dois only. iirc 10.1101 specifies cold spring harbor papers specifically (something like that).